cv19ResSupportV3.Tests/V3/Gateways/ResidentGatewayTest.cs

```diff
@@ -5,89 +5,477 @@
+
+            // act
+            var duplicateResidentId = _classUnderTest.FindResident(searchParameters);
+            bool isResidentDuplicate = duplicateResidentId != null;
+
+            // assert
+            isResidentDuplicate.Should().BeTrue();
+            duplicateResidentId.Should().Be(uprnMatchResidentId);
         }
 
-        [Test]
-        public void FindResidentWithDobandNameReturnsTheResidentIdIfItExists()
+        // If NHS number is empty or null or does not match &
+        // If Uprn is empty or null or does not match &
+        // If any of the Dob fields are missing (The focus here is on the Uprn paths, not the dob), then:
+        // Resident is considered unique even when the First Name, Last Name & Nonempty Uprn match.
+
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase(" ")]
+        [TestCase("uprn")]
+        public void FindResidentReturnsNoMatchWhenNoNhsNumberAndUprnNumberAndAnyOfTheDobFieldsArePresentForAllEmptyCasesOfUprn(string testCaseUprn)
         {
-            var existingResident = new ResidentEntity
+            // arrange
+            var notMatchingNhsNumber = _faker.Random.Hash();
+            var matchingFirstName = _faker.Random.Hash();
+            var matchingLastName = _faker.Random.Hash();
+            var matchingEmptyUprn = testCaseUprn;
+            // not even setting Dob fields (null by default)
+
+            var searchParameters = new FindResident
             {
-                FirstName = "FirstName",
-                LastName = "LastName",
-                DobDay = "Day",
-                DobMonth = "Month",
-                DobYear = "Year"
+                NhsNumber = notMatchingNhsNumber,
+                Uprn = matchingEmptyUprn,
+                FirstName = matchingFirstName,
+                LastName = matchingLastName,
             };
 
-            var findResidentCommand = new FindResident
+            var existingResidentNoUprnMatch = new ResidentEntity
             {
-                Uprn = "uprn",
-                FirstName = "FirstName",
-                LastName = "LastName",
-                DobDay = "Day",
-                DobMonth = "Month",
-                DobYear = "Year"
+                NhsNumber = _faker.Random.Hash(),
+                Uprn = _faker.Random.Hash(), // no uprn match
+                FirstName = matchingFirstName,
+                LastName = matchingLastName
+            };
+
+            var existingResidentEmptyUprnMatch = new ResidentEntity
+            {
+                NhsNumber = _faker.Random.Hash(),
+                Uprn = matchingEmptyUprn, // matching empty uprn
+                FirstName = matchingFirstName,
+                LastName = matchingLastName
+            };
+
+            // testing a case path, where uprn is not empty, but still does not match
+            if (matchingEmptyUprn == "uprn") existingResidentEmptyUprnMatch.Uprn = _faker.Random.Hash();
```

Comment:
Same as the test above

Reply:
Same answer as for the first 2 cases of this 😁